test/scholar/metrics_test.exs

```diff
@@ -1,5 +1,22 @@
 defmodule Scholar.MetricsTest do
   use ExUnit.Case, async: true
 
+  alias Scholar.Metrics
   doctest Scholar.Metrics
+
+  @y_true Nx.tensor([1.0, 1.0, 2.0, 2.0, 3.0, 3.0], type: {:f, 32})
+  @y_pred Nx.tensor([1.0, 2.0, 1.0, 3.0, 3.0, 3.0], type: {:f, 32})
+  @cm Nx.tensor(
+        [
+          [1, 1, 0],
+          [1, 0, 1],
+          [0, 0, 2]
+        ],
+        type: {:s, 64}
+      )
```

Comment:
You can cover this with a doctest by adding the result below

```
iex> Scholar.Metrics.confusion_matrix(y_true, y_pred)
```

in function docs. See the other metrics :)

Reply:
Oh, I forgot to mention that I tried to write the test here, but I encountered some weird errors. It works well if I only return a single Nx tensor, but it failed to parse the result when a tuple is returned. Not sure about the reason...

```
1) doctest Scholar.Metrics.confusion_matrix/3 (2) (Scholar.MetricsTest)
     test/scholar/metrics_test.exs:4
     Doctest did not compile, got: (SyntaxError) lib/scholar/metrics.ex:346:3: syntax error before: '['
          |
      346 |   [
          |   ^
     doctest:
       iex> y_true = Nx.tensor([1.0, 1.0, 2.0, 2.0, 3.0, 3.0], type: {:f, 32})
       iex> y_pred = Nx.tensor([1.0, 2.0, 1.0, 3.0, 3.0, 3.0], type: {:f, 32})
       iex> Scholar.Metrics.confusion_matrix(y_true, y_pred)
       {#Nx.Tensor<
         s64[3][3]
         [
           [1, 1, 0],
           [1, 0, 1],
           [0, 0, 2]
         ]
       >, 
       #Nx.Tensor<
         f32[3]
         [1.0, 2.0, 3.0]
       >}
```

But since we're going to ask the user to provide `:num_classes` in opts, this will not be a problem. :)